Issue with next-plugin-preact

An issue [vercel/next.js#26826](https://github.com/vercel/next.js/issues/26826) was brought up in the next.js repo where using Next.js 11 with webpack 5 seems to break the using-preact example.
``` info  - Using webpack 5. Reason: Enabled by default https://nextjs.org/docs/messages/webpack5
Error: Cannot find module 'webpack/lib/dependencies/ConstDependency'
Require stack:
- /home/projects/nextjs-preact-webpack5-a4fygy/node_modules/@prefresh/webpack/src/index.js
- /home/projects/nextjs-preact-webpack5-a4fygy/node_modules/@prefresh/next/src/index.js
- /home/projects/nextjs-preact-webpack5-a4fygy/node_modules/next-plugin-preact/index.js
- /home/projects/nextjs-preact-webpack5-a4fygy/next.config.js
- /home/projects/nextjs-preact-webpack5-a4fygy/node_modules/next/dist/next-server/server/config.js
- /home/projects/nextjs-preact-webpack5-a4fygy/node_modules/next/dist/server/next.js
- /home/projects/nextjs-preact-webpack5-a4fygy/node_modules/next/dist/server/lib/start-server.js
- /home/projects/nextjs-preact-webpack5-a4fygy/node_modules/next/dist/cli/next-dev.js
- /home/projects/nextjs-preact-webpack5-a4fygy/node_modules/next/dist/bin/next
    at Function.Module._resolveFilename (https://nextjs-preact-webpack5-a4fygy.jw.staticblitz.com/blitz.70a1ff62a0f0131ac0df3c30faf487fef744ac90.js:6:189728)
    at Function.mod._resolveFilename (/home/projects/nextjs-preact-webpack5-a4fygy/node_modules/next/dist/build/webpack/require-hook.js:4:1855)
    at Function.Module._resolveFilename (/home/projects/nextjs-preact-webpack5-a4fygy/node_modules/module-alias/index.js:49:29)
    at Function.Module._load (https://nextjs-preact-webpack5-a4fygy.jw.staticblitz.com/blitz.70a1ff62a0f0131ac0df3c30faf487fef744ac90.js:6:187342)
    at Module.require (https://nextjs-preact-webpack5-a4fygy.jw.staticblitz.com/blitz.70a1ff62a0f0131ac0df3c30faf487fef744ac90.js:6:190516)
    at i (https://nextjs-preact-webpack5-a4fygy.jw.staticblitz.com/blitz.70a1ff62a0f0131ac0df3c30faf487fef744ac90.js:6:477111)
    at _0x52c652 (https://nextjs-preact-webpack5-a4fygy.jw.staticblitz.com/blitz.70a1ff62a0f0131ac0df3c30faf487fef744ac90.js:11:190288)
    at ReloadPlugin.webpack5 (/home/projects/nextjs-preact-webpack5-a4fygy/node_modules/@prefresh/webpack/src/index.js:60:29)
    at ReloadPlugin.apply (/home/projects/nextjs-preact-webpack5-a4fygy/node_modules/@prefresh/webpack/src/index.js:170:14)
    at createCompiler (/home/projects/nextjs-preact-webpack5-a4fygy/node_modules/next/dist/compiled/webpack/bundle5.js:141349:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/home/projects/nextjs-preact-webpack5-a4fygy/node_modules/@prefresh/webpack/src/index.js',
    '/home/projects/nextjs-preact-webpack5-a4fygy/node_modules/@prefresh/next/src/index.js',
    '/home/projects/nextjs-preact-webpack5-a4fygy/node_modules/next-plugin-preact/index.js',
    '/home/projects/nextjs-preact-webpack5-a4fygy/next.config.js',
    '/home/projects/nextjs-preact-webpack5-a4fygy/node_modules/next/dist/next-server/server/config.js',
    '/home/projects/nextjs-preact-webpack5-a4fygy/node_modules/next/dist/server/next.js',
    '/home/projects/nextjs-preact-webpack5-a4fygy/node_modules/next/dist/server/lib/start-server.js',
    '/home/projects/nextjs-preact-webpack5-a4fygy/node_modules/next/dist/cli/next-dev.js',
    '/home/projects/nextjs-preact-webpack5-a4fygy/node_modules/next/dist/bin/next'
  ]
} 
```

I attempted to install webpack5 on its own as a dependency to the package,json file in [vercel/next.js#26864](https://github.com/vercel/next.js/pull/26864) but one of the lead developers suggested that the @prefresh/webpack package is using webpack internals that can't be required as they're no longer available.

This issue was mentioned here as well: [preactjs/next-plugin-preact#34](https://github.com/preactjs/next-plugin-preact/issues/34)

--- a/.changeset/gold-planes-shave.md
+++ b/.changeset/gold-planes-shave.md
@@ -1,0 +1,5 @@
+---
+'@prefresh/webpack': patch
+---
+
+Remove obsolete webpack deps

--- a/packages/webpack/src/index.js
+++ b/packages/webpack/src/index.js
@@ -56,10 +56,7 @@ class ReloadPlugin {
     });
   }
 
-  webpack5(compiler) {
-    const ConstDependency = require('webpack/lib/dependencies/ConstDependency');
-
-    const RuntimeGlobals = require('webpack/lib/RuntimeGlobals');
+  webpack5(compiler, RuntimeGlobals) {
     const PrefreshRuntimeModule = require('./utils/Runtime');
 
     compiler.hooks.compilation.tap(
@@ -70,11 +67,6 @@ class ReloadPlugin {
         }
 
         injectRefreshFunctions(compilation);
-
-        compilation.dependencyTemplates.set(
-          ConstDependency,
-          new ConstDependency.Template()
-        );
 
         compilation.hooks.additionalTreeRuntimeRequirements.tap(
           NAME,
@@ -167,7 +159,8 @@ class ReloadPlugin {
             callback
           );
         });
-        this.webpack5(compiler);
+        
+        this.webpack5(compiler, compiler.webpack ? compiler.webpack.RuntimeGlobals : webpack.RuntimeGlobals);
         break;
       }
       default: {

